### ISSUE TITLE
[Grid - Grouping] Checkboxes in stories Selection and Both do not have sufficient names

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3391]
### 😯 Current Behavior

In stories [Selection](https://storybook.sanomalearning.design/?path=/story/grid-grouping--selection) and [Both](https://storybook.sanomalearning.design/?path=/story/grid-grouping--both) checkboxes that allow to check whole group do not have accessible name and are announced as "checkbox, unchecked".

Additionally checkboxes that allow users to check only one row are named "Select row" without information what row can be selected. This may be confusing to users who would like to get full information about row that can be selected.

<img width="1195" height="356" alt="Image" src="https://github.com/user-attachments/assets/fe3cf7b0-0f0e-4216-a04e-b12c1d748565" />

https://github.com/user-attachments/assets/3433fac1-20b8-4cf5-b110-465370abdc16

### 🤔 Expected Behavior/possible solution?

Checkboxes that are used to select whole group should have names like: "Select group NAME OF GROUP" or "NAME OF GROUP select group".

Checkboxes that are used to select individual rows should have more unique names like: "Select row STUDENT'S NAME".

Names can be added to `<input type="checkbox">` using `aria-labelledby` or `aria-label`. `aria-labelledby` is the most recommended solution in this case.

### 🖥️ Steps to Reproduce

1. Go to https://storybook.sanomalearning.design/?path=/story/grid-grouping--selection or https://storybook.sanomalearning.design/?path=/story/grid-grouping--both
2. With screen reader turned on place your focus on each checkbox 
3. Check if every checkbox has an unique name that tells what is the purpose of checkbox

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/grid-grouping--both

### ⚖️ Impact

This blocks our release or stops us from using this component

### Framework

Other

### What browsers are you seeing the problem on?

Other, All

### Component version

v0.7.2

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
